mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOpInterfaces.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) Intel
+
+//===--- SYCLOpInterfaces.h -----------------------------------------------===//
+//
+// MLIR-SYCL is under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_SYCL_OPS_INTERFACES_H_
+#define MLIR_SYCL_OPS_INTERFACES_H_
+
+#include <type_traits>
+
+#include "mlir/Dialect/SYCL/IR/SYCLOpsTypes.h"
+
+#include "mlir/Dialect/SYCL/IR/SYCLOpInterfaces.h.inc"
+
+namespace mlir {
```

Comment:
`mlir/include/mlir/Dialect/Affine/IR/AffineMemoryOpInterfaces.h` puts the `.h.inc` file in namespace mlir.

```suggestion
namespace mlir {
#include "mlir/Dialect/SYCL/IR/SYCLOpInterfaces.h.inc"
```

Reply:
[This line](https://github.com/victor-eds/llvm/blob/7905cf048d50f3f9d21b440d8a3710a6cb3c0e75/mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOpInterfaces.td#L17) makes that unnecessary:
```
class SYCL_OpInterface<string name> : OpInterface<name> {
  let cppNamespace = "::mlir::sycl";
}
```